prometheus-client/prometheus-query-api-spec.yaml

```diff
@@ -0,0 +1,140 @@
+# This is a file maintained by the rhsm-subscriptions project and describes
+# a portion of the Prometheus Query API.
```

Comment:
Not as part of this PR, but after things are working can the query API team take over this file?

Reply:
The problem right now is that this definition contains some hacks to work around a limitation of the OpenAPI Java code generator that we use. It doesn't properly generate code to support dynamic response objects required by the GET /query endpoint. Even if they took this over and defined it the 'correct' way, until that support is added to the generator, we wouldn't really be able to use it.

That being said, it might be a good starting point.

This was really just hacked together to get us going. I'm really not a fan of the end result. It is a little hacky, but works.